amity/test_amity.py

```diff
@@ -0,0 +1,275 @@
+        self.assertEquals(len(amity.people), 0)
+        self.assertEquals(len(amity.fellows), 0)
+        self.assertEquals(len(amity.staff), 0)
+
+    # create room method test functionality begins here
+    def test_returns_error_when_non_string_is_addded(self):
+        amity = Amity()
+        self.assertEqual(amity.create_room(
+            0, 2), 'Error. Invalid room type initial.',
+            msg='Room name and initial must only be strings.')
+        self.assertEqual(amity.create_room('w', 'WorkSpace'),
+                         'Error. Invalid room type initial.',
+                         msg='Enter O or L for room type inital.')
+
+    def test_create_room_method(self):
+        amity = Amity()
+        with patch("amity.rooms.room.Office"):
+            amity.create_room("o", "Lime")
+            self.assertIn("Lime", amity.offices['available'])
+        with patch("amity.rooms.room.LivingSpace"):
+            amity.create_room('l', 'python')
+            self.assertIn('Python', amity.living_spaces['available'])
+
+    def test_create_room_increases_rooms_list_by_one(self):
+        amity = Amity()
+        room_count_before = len(amity.rooms)
+        with patch("amity.rooms.room.Office"):
+            amity.create_room("o", "Lime")
+            room_count_after = len(amity.rooms)
+            self.assertEquals((room_count_after - room_count_before), 1)
+            amity.create_room('o', 'oculus')
+            room_count_after_two = len(amity.rooms)
+            self.assertAlmostEquals(
+                (room_count_after_two - room_count_before), 2)
+
+    def test_living_space_can_only_be_created_once(self):
+        amity = Amity()
+        with patch('amity.rooms.room.LivingSpace'):
+            amity.create_room('l', 'ruby')
+            result = amity.create_room('l', 'ruby')
+            self.assertEqual(result, 'Room already exists.')
+
+    def test_office_can_only_be_created_once(self):
+        amity = Amity()
+        with patch('amity.rooms.room.Office'):
+            amity.create_room('o', 'orange')
+            result = amity.create_room('o', 'orange')
+            self.assertEqual(result, 'Room already exists.')
+
+    def test_room_creation_when_successful(self):
+        amity = Amity()
+        with patch('amity.rooms.room.Office'):
+            result = amity.create_room('o', 'krypton')
+            self.assertEqual(result, 'Room Krypton created.')
+        amity = Amity()
+        with patch('amity.rooms.room.LivingSpace'):
+            result = amity.create_room('l', 'haskell')
+            self.assertEqual(result, 'Room Haskell created.')
+
+    # Print allocations method test begins here
+    def test_returns_no_allocations_if_no_rooms_created(self):
+        amity = Amity()
+        self.assertEqual(amity.print_allocations(),
+                         'Error. No rooms within system.')
+    # add_person method testing begins here
+
+    def test_returns_error_if_no_rooms_within_system(self):
```

Comment:
This is a different validation. I don't want to accumulate unallocated people within my system but instead return an error that prompts them to add rooms first.